test/fuzz/Rebalancer.t.sol

```diff
@@ -0,0 +1,636 @@
+            ((guaranteedRate1 / 2) * decimalsB) / 1e18
+        );
+
+        vm.startPrank(governor);
+        rebalancer.setOrder(address(eurY), address(eurB), subsidyBudget1 / 3, guaranteedRate1);
+        vm.stopPrank();
+        (a, b) = rebalancer.orders(address(eurY), address(eurB));
+        assertEq(a, subsidyBudget1 / 3);
+        assertEq(b, guaranteedRate1);
+        assertEq(rebalancer.budget(), subsidyBudget + subsidyBudget1 / 3);
+        assertEq(
+            rebalancer.getGuaranteedAmountOut(address(eurY), address(eurB), decimalsY),
+            (guaranteedRate1 * decimalsB) / 1e18
+        );
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        RECOVER                                                     
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    function test_Recover_RevertWhen_NonGovernorOrGuardian() public {
+        vm.expectRevert(Errors.NotGovernorOrGuardian.selector);
+        rebalancer.recover(address(eurA), 100, address(governor));
+    }
+
+    function test_Recover_RevertWhen_InvalidParam() public {
+        vm.startPrank(governor);
+        vm.expectRevert();
+        rebalancer.recover(address(eurA), 100, address(governor));
+        vm.expectRevert(Errors.InvalidParam.selector);
+        rebalancer.recover(address(agToken), 100, address(governor));
+        vm.stopPrank();
+    }
+
+    function testFuzz_Recover(uint256 amount) public {
+        amount = bound(amount, 10 ** 9, 10 ** 27);
+        deal(address(eurB), address(rebalancer), amount);
+        vm.startPrank(governor);
+        rebalancer.recover(address(eurB), amount / 2, address(governor));
+        vm.stopPrank();
+        assertEq(eurB.balanceOf(address(governor)), amount / 2);
+        vm.startPrank(governor);
+        deal(address(agToken), address(rebalancer), amount);
+        rebalancer.setOrder(address(eurA), address(eurB), amount / 2, BASE_18 / 100);
+        vm.stopPrank();
+        (uint256 a, uint256 b) = rebalancer.orders(address(eurA), address(eurB));
+        assertEq(a, amount / 2);
+        assertEq(b, BASE_18 / 100);
+        vm.startPrank(governor);
+        vm.expectRevert(Errors.InvalidParam.selector);
+        rebalancer.recover(address(agToken), amount - 1, address(governor));
+        vm.stopPrank();
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                       QUOTE IN                                                     
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    function testFuzz_QuoteInWithoutSubsidy(uint256 multiplier, uint256 swapMultiplier) public {
```

Comment:
Let's also try by setting fees (of the order of current transmuter fees) and do the actual swaps vs the quote we get at the end. To see when it starts to fail

Reply:
Could do this, I feel the logic is pretty well understood on when this is actually going to be different and fail, like i'm not sure we'd learn a ton from a test here